store/driver/txn/batch_getter.go

```diff
@@ -16,11 +16,54 @@ package txn
 
 import (
 	"context"
+	"unsafe"
 
-	"github.com/pingcap/errors"
 	"github.com/pingcap/tidb/kv"
+	tikverr "github.com/tikv/client-go/v2/error"
+	"github.com/tikv/client-go/v2/txnkv/transaction"
 )
 
+// tikvBatchGetter is the BatchGetter struct for tikv
+type tikvBatchGetter struct {
+	tidbBatchGetter BatchGetter
+}
+
+func (b tikvBatchGetter) BatchGet(ctx context.Context, keys [][]byte) (map[string][]byte, error) {
+	// toTiDBKeys
+	kvKeys := *(*[]kv.Key)(unsafe.Pointer(&keys))
+	vals, err := b.tidbBatchGetter.BatchGet(ctx, kvKeys)
+	return vals, err
+}
+
+// tikvBatchBufferGetter is the BatchBufferGetter struct for tikv
+type tikvBatchBufferGetter struct {
+	tidbMiddleCache Getter
+	tidbBuffer      BatchBufferGetter
+}
+
+func (b tikvBatchBufferGetter) Get(k []byte) ([]byte, error) {
+	// Get from buffer
+	val, err := b.tidbBuffer.Get(context.TODO(), k)
+	if err == nil || !kv.IsErrNotFound(err) || b.tidbMiddleCache == nil {
+		if kv.IsErrNotFound(err) {
+			err = tikverr.ErrNotExist
+		}
+		return val, err
+	}
+	// Get from middle cache
+	val, err = b.tidbMiddleCache.Get(context.TODO(), k)
+	if err == nil {
+		return val, err
+	}
+	// TiDB err NotExist to TiKV err NotExist
```

Comment:
And could you please explain why we need to use TiKV's NotExist here?

Reply:
Because this function will be called in client-go, and returns `err`. In client-go, the following code:
```rust
if !tikverr.IsErrNotFound(err) {
        return nil, errors.Trace(err)
}
```
If we don't convert `err` to TiKV NotExist, it will return `nil` directly.

Reply:
Yes I understand it. I just recommend you briefly explain them in the code comments 😂

Reply:
Oh, sorry.😂 I'll add the comments later.